app/controllers/DirectoryController.scala

```diff
@@ -18,14 +19,17 @@ class DirectoryController @Inject()(branchRepository: BranchRepository, director
 
     (for {
       projectId <- params.lift(0)
-      branchName <- params.lift(1)
+      paramsBranchName <- params.lift(1)
       relativePath <- params.lift(2)
 
+      project <- projectRepository.findById(projectId)
+      branchName = if (paramsBranchName.equals("")) project.stableBranch else paramsBranchName
+
       branchId <- branchRepository.findByProjectIdAndName(projectId, branchName).map(_.id)
       directory <- directoryRepository.findByBranchIdAndRelativePath(branchId, relativePath)
-      pagesPaths = pageRepository.findAllByDirectoryId(directory.id)
+      pagesPaths = pageRepository.findAllByDirectoryId(directory.id).map(page => page.copy(path = getCorrectedPath(page.path, project)))
```

Comment:
This logic should not be in the controller. You should instead call pageService.getAllPagePaths and in the service, do whatever is necessary to retrieve the page paths